src/lib.rs

```diff
@@ -207,6 +207,17 @@ impl<T: SubCommand> SubCommands for T {
     const COMMANDS: &'static [&'static CommandInfo] = &[T::COMMAND];
 }
 
+/// A `HelpMessage` implementation that provides a help/usage message corresponding
+/// to the type's `FromArgs` implementation.
+pub trait HelpMessage: FromArgs {
```

Comment:
Why an additional trait instead of adding a method to `FromArgs`?

Reply:
The help message only exists for `struct`s, not for `enum`s, so it's not implemented for every type that implements `FromArgs`